distmv/normal.go

```diff
@@ -75,9 +75,20 @@ func NewNormal(mu []float64, sigma mat64.Symmetric, src *rand.Rand) (*Normal, bo
 // ConditionNormal returns {nil, false} if there is a failure during the update.
 // Mathematically this is impossible, but can occur with finite precision arithmetic.
 func (n *Normal) ConditionNormal(observed []int, values []float64, src *rand.Rand) (*Normal, bool) {
+	if len(observed) == 0 {
+		panic("normal: no observed values")
+	}
 	if len(observed) != len(values) {
 		panic("normal: input slice length mismatch")
 	}
+	if len(observed) == n.Dim() {
+		panic("normal: all dimensions observed")
```

Comment:
Wouldn't it be more informative to have the distinct error messages (rather than weaken the text)? We have to check both individually anyway, so seems like we might as well give an informative error message.